### PR TITLE
Support `AWSCredentialsProviderChain` and `STSAssumeRoleSessionCredentialsProvider`

### PR DESCRIPTION
**Pull request by cw-ozaki**
Support few providers.

http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/AWSCredentialsProviderChain.html
http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/STSAssumeRoleSessionCredentialsProvider.html

### CODE REVIEW

**Comment by cw-junichikato** [comment]
新規追加したクラスへの暗黙的型変換を以下に追加してもらえますか？

https://github.com/sisioh/aws4s/blob/feature/add-providers/aws4s-core/src/main/scala/org/sisioh/aws4s/Implicits.scala

**Comment by cw-ozaki** [comment]
@cw-junichikato 
追加しました！

File: aws4s-core/src/main/scala/org/sisioh/aws4s/core/auth/RichAWSCredentialsProviderChain.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.sisioh.aws4s.core.auth
+
+import com.amazonaws.auth.{ AWSCredentials, AWSCredentialsProviderChain }
+import org.sisioh.aws4s.PimpedType
+
+class RichAWSCredentialsProviderChain(val underlying: AWSCredentialsProviderChain)
```
Comment (cw-junichikato) [inline aws4s-core/src/main/scala/org/sisioh/aws4s/core/auth/RichAWSCredentialsProviderChain.scala]:
```scala
object AWSCredentialsProviderChainFactory {
   def create(): AWSCredentialsProviderChain = new AWSCredentialsProviderChain()
   // 他のコンストラクタ分も書く
   // def create(...): AWSCredentialsProviderChain = new AWSCredentialsProviderChain(...)
}
```

Comment (cw-ozaki) [inline aws4s-core/src/main/scala/org/sisioh/aws4s/core/auth/RichAWSCredentialsProviderChain.scala]: @cw-junichikato 
必要そうなので上部に追加しました。

**Pull request merged**